test/test_operations.py

```diff
@@ -298,7 +298,9 @@ def assertTensorsEqual(a, b):
             if diff.is_signed() and diff.dtype != torch.int8:
               diff = diff.abs()
             max_err = diff.max()
-            self.assertLessEqual(max_err, prec, message)
+            if not (a.isnan().any() and b.isnan().any() and
+                    torch.isnan(max_err)):
+              self.assertLessEqual(max_err, prec, message)
```

Comment:
why do we check for `torch.isnan(max_err)`? If there is nan in the input, we verify 
1. `a.isnan()` == `b.isnan()`
2. non-nan part of a and b matches.

If you just skip check when there is `nan`, test will be a no-op if `a` or `b` contains `nan` I think.

Reply:
`self.assertLessEqual()` fails on inputs containing a `nan`. The updated implementation does an assertTrue() on a.isnan() == b.isnan() for tensors with `nan` entries. Wdyt?

Reply:
I think we should move the logic between line 285->296 out of the `a.is_floating_point()` if statement. That will handle the `nan` and `inf` comparison and replace them to be 0. One thing you might want to check is whether `0` replacement would work for all types, for example whether assign 0 to a boolean tensor will throw an error.

Reply:
I think this part is still missing. You shouldn't need to do `nan` check here if `a.is_floating_point()` is removed.